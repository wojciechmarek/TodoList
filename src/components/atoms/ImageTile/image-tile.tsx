import { ImageTileProps } from './image-tile.interface';
import { ImageTileContainer } from './image-tile.styled';

export const ImageTile = ({ text, handleThemeClick }: ImageTileProps) => {
  return (
    <ImageTileContainer onClick={handleThemeClick}>{text}</ImageTileContainer>
  );
};
