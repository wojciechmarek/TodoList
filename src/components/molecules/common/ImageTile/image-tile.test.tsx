import { configure, render } from '@testing-library/react';
import { ImageTileProps } from './image-tile.interface';
import { ImageTile } from './image-tile';

describe('ImageTile', () => {
  let imageTileProps: ImageTileProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    imageTileProps = {
      src: 'https://via.placeholder.com/150',
      text: 'text',
      isActive: false,
      alt: 'alt',
      handleThemeClick: jest.fn(),
    };
  });

  it('should render', () => {
    const { baseElement } = render(<ImageTile {...imageTileProps} />);
    expect(baseElement).toBeTruthy();
  });
});
