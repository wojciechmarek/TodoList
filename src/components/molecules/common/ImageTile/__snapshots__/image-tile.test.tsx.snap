// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageTile should match the snapshot 1`] = `
.emotion-0 {
  margin: 0.75em;
  height: 5em;
  width: 10em;
  display: grid;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--theme-tile-button-bg-color);
  border-radius: 0.5em;
  color: var(--button-font-color);
  font-weight: 800;
  border: var(--standard-border);
  border-radius: var(--standard-border-radius);
  box-shadow: var(--container-box-shadow);
  cursor: pointer;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  -webkit-filter: none;
  filter: none;
}

.emotion-0:hover {
  -webkit-filter: brightness(0.75);
  filter: brightness(0.75);
}

<div
  className="emotion-0"
  data-testid="tile"
  onClick={[MockFunction]}
>
  text
</div>
`;
